ui/test/nuxeo-document-layout.test.js

```diff
@@ -0,0 +1,309 @@
+    // we need to ad a listener for this event before hand, because we do not yet have the layout stamped,
+    // and after `buildLayout` it can be too late to catch the event.
+    const layoutChanged = new Promise((resolve) => {
+      document.addEventListener('document-layout-changed', (e) => resolve(e));
+    });
+    documentLayout = await buildLayout();
+    // check that the event was fired from the document layout
+    const event = await layoutChanged;
+    expect(event.target).to.equal(documentLayout);
+    // ensure that the event bubbles and is composed
+    expect(event.bubbles).to.be.true;
+    expect(event.composed).to.be.true;
+    // verify event details
+    expect(event.detail).to.exist;
+    expect(event.detail.element).to.equal(documentLayout.element);
+    expect(event.detail.layout).to.equal(documentLayout.layout);
+  });
+
+  test('Should reload when layout name changes', async () => {
+    documentLayout = await buildLayout();
+    expect(documentLayout.element).to.exist;
+    expect(documentLayout.element.tagName).to.equal('NUXEO-TEST-EDIT-LAYOUT');
+    // check that the layout updates when the layout name changes
+    documentLayout.layout = 'view';
+    const event = await waitForEvent(documentLayout, 'document-layout-changed');
+    expect(event.detail).to.exist;
+    const { element } = event.detail;
+    expect(element).to.exist;
+    expect(element.tagName).to.equal('NUXEO-TEST-VIEW-LAYOUT');
+    // assert layout integrity
+    const { children } = element.shadowRoot;
+    expect(children).to.have.lengthOf(3);
+    expect(children[0].tagName).to.equal('STYLE');
+    expect(children[1].tagName).to.equal('H1');
+    expect(children[2].tagName).to.equal('DIV');
+  });
+
+  test('Should reload when document changes', async () => {
+    documentLayout = await buildLayout();
+    expect(documentLayout.element).to.exist;
+    expect(documentLayout.element.tagName).to.equal('NUXEO-TEST-EDIT-LAYOUT');
+    // check that the layout updates when the document changes
+    documentLayout.document = { type: 'File', uid: '12ae', properties: {} };
+    const event = await waitForEvent(documentLayout, 'document-layout-changed');
+    expect(event.detail).to.exist;
+    expect(event.detail.element).to.be.undefined;
+    // check 404 for unexisting layout
+    assertNotFound();
+  });
+
+  test('Should reload when href template changes', async () => {
+    documentLayout = await buildLayout();
+    expect(documentLayout.element).to.exist;
+    expect(documentLayout.element.tagName).to.equal('NUXEO-TEST-EDIT-LAYOUT');
+    // check that the layout updates when the document changes
+    documentLayout.hrefTemplate = '../dummy-layout.html';
+    const event = await waitForEvent(documentLayout, 'document-layout-changed');
+    expect(event.detail).to.exist;
+    const { element } = event.detail;
+    expect(element).to.exist;
+    expect(element.tagName).to.equal('DUMMY-LAYOUT');
```

Comment:
nit: if you are "one-liner" guy
`expect(event.detail).to.exist.and.to.have.property('tagName').which.is.equal('DUMMY-LAYOUT')`

Reply:
Not sure we can do a one-liner here (that's readable), since we're missing the `element` there, which what has the tagName.